hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -75,13 +76,13 @@
   public HoodieLogFileReader(FileSystem fs, HoodieLogFile logFile, Schema readerSchema, int bufferSize,
                              boolean readBlockLazily, boolean reverseReader) throws IOException {
     FSDataInputStream fsDataInputStream = fs.open(logFile.getPath(), bufferSize);
-    if (FSUtils.isGCSInputStream(fsDataInputStream)) {
-      this.inputStream = new TimedFSDataInputStream(logFile.getPath(), new FSDataInputStream(
-          new BufferedFSInputStream((FSInputStream) ((
-              (FSDataInputStream) fsDataInputStream.getWrappedStream()).getWrappedStream()), bufferSize)));
-    } else if (fsDataInputStream.getWrappedStream() instanceof FSInputStream) {
+    if (fsDataInputStream.getWrappedStream() instanceof FSInputStream) {
       this.inputStream = new TimedFSDataInputStream(logFile.getPath(), new FSDataInputStream(
           new BufferedFSInputStream((FSInputStream) fsDataInputStream.getWrappedStream(), bufferSize)));
+    } else if (FSUtils.isGCSFileSystem(fs)) {
```

Comment:
This piece of logic still slightly worries me a bit.

I would also check fsDataInputStream.getWrappedStream().getWrappedStream() could be cast to FSInputStream and fallback to 'this.inputStream = fsDataInputStream' if it can't be. But I also would log a warning in this case as it would be unexpected situation that we may want to dig deeper in the future.

thoughts?

Reply:
@nsivabalan @bvaradar could you please clarify why we need this casting business here in the first place. With @vburenin . If we can simplify all this, we should

Reply:
I had two options. either check two conditions like
```
if(fsDataInputStream.getWrappedStream() instance of FSDataInputStream && ((FSDataInputStream)fsDataInputStream.getWrappedStream()).getWrappedStream() instance of FSInputStream)
```
or resort to casting and if exception if thrown fallback to original dataInputStream. I have gone w/ 2nd approach.